ebu_tt_live/node/handover.py

```diff
@@ -60,23 +60,31 @@ def process_document(self, document, **kwargs):
         """
         emit = False
 
-        if self.check_document(document=document) is True \
-                and document.authors_group_identifier == self._authors_group_identifier \
-                and document.authors_group_control_token is not None:
-            if self._current_token is None or self._current_token < document.authors_group_control_token:
-                # Switch input
-                self._current_selected_input_sequence_id = document.sequence_identifier
-                emit = True
-            elif self._current_selected_input_sequence_id == document.sequence_identifier:
-                emit = True
-
-        if emit is True:
-            # Update token
-            self._current_token = document.authors_group_control_token  # So are we going to error here?
-            self._last_sequence_number += 1
-
+        if self.is_document(document):
+            if self.check_if_document_seen(document=document) is True \
+                    and document.authors_group_identifier == self._authors_group_identifier \
+                    and document.authors_group_control_token is not None:
+                if self._current_token is None or self._current_token < document.authors_group_control_token:
+                    # Switch input
+                    self._current_selected_input_sequence_id = document.sequence_identifier
+                    emit = True
+                elif self._current_selected_input_sequence_id == document.sequence_identifier:
+                    emit = True
+
+            if emit is True:
+                # Update token
+                self._current_token = document.authors_group_control_token  # So are we going to error here?
```

Comment:
What's happening with this comment? Looks like we cannot get here if the  authors group control token is absent. Or is there another case to consider?